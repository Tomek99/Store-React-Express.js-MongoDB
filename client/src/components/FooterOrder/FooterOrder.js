import React, { useState } from "react";
import styles from "./FooterOrder.module.scss";
import { BsHeadphones } from "react-icons/bs";
import BtnMore from "../Buttons/BtnMore/BtnMore";
import { Link } from "react-router-dom";
function FooterOrder() {
  const [isActive, setIsActive] = useState(false);

  function handleBtn() {
    setIsActive(!isActive);
  }
  return (
    <div className={styles.FooterOrder}>
      <div className={styles.flexColumn}>
        <div className={styles.columnLeft}>
          <div className={styles.divAsk}>
            <p>Do you have any questions?</p>
            <p>
              <span
                style={{ display: "flex", alignItems: "center", gap: "5px" }}
              >
                <BsHeadphones size={20} /> 24 242 24 24
              </span>
            </p>
          </div>
          <div className={styles.divStatue}>
            <Link>Statute</Link>
            <Link>Privacy policy</Link>
          </div>
        </div>
        <div className={styles.columnRight}>
          <p className={styles.dataOwner}>
            The administrator of your personal data is coffeShop sp. z o.o.
            based in Jaros≈Çaw, 25 XYZ St., 00-000 Jaroslaw.
            <BtnMore handleBtn={handleBtn} />
          </p>
          {isActive ? (
            <p className={styles.moreInfo}>
              We will process your personal data to register and operate your
              account in our store or to process orders you place. If you give
              the appropriate consents, we will also use your data for profiling
              and sending you marketing content. Remember that you can withdraw
              your consent at any time. The security of your data is very
              important to us. You can find detailed rules for processing your
              personal data and the rights you have in this regard in ourPrivacy
              Policy
            </p>
          ) : null}
        </div>
      </div>
    </div>
  );
}

export default FooterOrder;
