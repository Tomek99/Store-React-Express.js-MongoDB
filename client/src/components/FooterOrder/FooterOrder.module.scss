.FooterOrder {
    position: relative;
    display: grid;
    grid-template-columns: 1fr;
    justify-items: center;
    align-items: center;
    font-size: 1.3rem;
    color: #fff;
    margin: 0 10px;

    .flexColumn {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        max-width: 1200px;
        width: 100%;
        padding: 12px 0 12px;
        border-top: var(--border);

        .columnLeft {
            .divAsk {
                display: flex;
                align-items: center;
                gap: 10px;
            }

            .divStatue {
                display: flex;
                gap: 10px;

                a {
                    color: #fff;

                    &:hover {
                        text-decoration: underline;
                    }
                }
            }
        }

        .columnRight {
            max-width: 360px;
            color: #ccc;

            .dataOwner {
                text-align: right;
            }

            .moreInfo {
                position: absolute;
                text-align: right;
                margin: 10px 5px;
            }
        }
    }
}

@media (max-width: 900px) {
    .flexColumn {
        flex-direction: column !important;
        justify-content: center;
        align-items: center;

        .columnLeft {
            .divStatue {
                margin-top: 5px;
                justify-content: center;
            }
        }
        .columnRight {
            margin-top: 10px;

            .dataOwner {
                text-align: center !important;
            }
            .moreInfo {
                transform: translateX(-20%);
                text-align: center !important;
            }
        }
    }
}
@media (max-width: 750px) {
    .moreInfo {
        transform: translateX(-10%) !important;
    }
}

@media (max-width: 500px) {
    .moreInfo {
        transform: translateX(-5%) !important;
    }
}

@media (max-width: 400px) {
    .moreInfo {
        transform: none !important;
    }
}
