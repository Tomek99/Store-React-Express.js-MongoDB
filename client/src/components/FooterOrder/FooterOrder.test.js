import { render, within, screen } from "@testing-library/react";
import FooterOrder from "./FooterOrder";
import { MemoryRouter } from "react-router-dom";

test("should render FooterOrder component", () => {
  const { container } = render(
    <MemoryRouter>
      <FooterOrder />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
