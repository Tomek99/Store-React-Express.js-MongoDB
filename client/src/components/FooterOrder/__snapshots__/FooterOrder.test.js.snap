// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FooterOrder component 1`] = `
<div>
  <div
    class="FooterOrder"
  >
    <div
      class="flexColumn"
    >
      <div
        class="columnLeft"
      >
        <div
          class="divAsk"
        >
          <p>
            Do you have any questions?
          </p>
          <p>
            <span
              style="display: flex; align-items: center; gap: 5px;"
            >
              <svg
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 3a5 5 0 0 0-5 5v1h1a1 1 0 0 1 1 1v3a1 1 0 0 1-1 1H3a1 1 0 0 1-1-1V8a6 6 0 1 1 12 0v5a1 1 0 0 1-1 1h-1a1 1 0 0 1-1-1v-3a1 1 0 0 1 1-1h1V8a5 5 0 0 0-5-5z"
                />
              </svg>
               24 242 24 24
            </span>
          </p>
        </div>
        <div
          class="divStatue"
        >
          <a
            href="/"
          >
            Statute
          </a>
          <a
            href="/"
          >
            Privacy policy
          </a>
        </div>
      </div>
      <div
        class="columnRight"
      >
        <p
          class="dataOwner"
        >
          The administrator of your personal data is coffeShop sp. z o.o. based in Jaros≈Çaw, 25 XYZ St., 00-000 Jaroslaw.
          <button
            class="BtnMore"
            type="button"
          >
            More
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;
